src/benchmarks/micro/libraries/System.Text.Encodings.Web/Perf.UrlEncoder.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Text.Encodings.Web;
+
+namespace System.Text.Tests
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class Perf_UrlEncoder
+    {
+        [Params("&lorem ipsum=dolor sit amet")]
+        public string sourceString;
```

Comment:
why is the `sourceString` different for all 3 types? if we want to have apples-to-apples comparison, it should be the same

Reply:
They are different encoders escaping different characters, so the variation in inputs exists merely to induce some escaping in all of them. The three types have very different implementations, so this is not (and should not be) attempting to compare them to each other.